api/src/main/java/bio/terra/tanagra/service/query/api/ApiConversionService.java

```diff
@@ -72,6 +72,7 @@ public EntityDataset convertEntityDataset(
         .orderByAttribute(orderByAttribute)
         .orderByDirection(convertOrderByDirection(apiEntityDataset.getOrderByDirection()))
         .filter(filter)
+        .pageSize(pageSize)
```

Comment:
It would be good to add some validation here to make sure the `limit` is a positive integer.